Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -192,6 +193,13 @@ private void OnGameChange(GameRunLevelChangedEvent ev)
         if (!CanSendInGame(message, shell, player))
             return;
 
+        if (HasComp<BorgChassisComponent>(source))
+        {
+            //Allows borgs in a critical state to speak
+            ignoreActionBlocker = true;
+        }
```

Comment:
this 100% should use an event to genericize this behavior rather than being tied to this function in chatsystem. Furthermore, it should probably be a generic component or something on SpeechComponent that allows this rather than being implicity tied to borgchassiscomponent.

Reply:
Alrighty, I've made it a generic component on SpeechComponent and applied it to Borgs, getting rid of the tie to BorgChassisComponent

Will look into using an event for it instead but any extra clarification/explanation on this is appreciated

Reply:
So after messing with it for an hour I'm not really sure how to implement an event for this behavior when all I really want it to do is done by this ChatSystem check. To me the behavior seems properly genericized by the addition of the new generic component that is applicable to other entities if it is so wished, and the only function of the new UnblockableSpeechComponent is just to make IC speech always send-able by certain entities, I am unsure of how else to further genericize it.

Am I missing something with this? Overall just asking for a short reconsideration on the first point, and if it is still necessary, would appreciate some explanation on what a good implementation of that would be and what the purpose of doing so is, thank you

Reply:
Alrighty! After awhile I've finally figured out how to make this use an event, which should mean both of these concerns are now addressed while maintaining the proper functionality! As a personal note, I ended up looking into the accent system for references of implementation.